master/buildbot/test/util/migration.py

```diff
@@ -79,9 +79,10 @@ def upgrade_thd(engine):
                 schema = migrate.versioning.schema.ControlledSchema(
                     engine, self.db.model.repo_path)
                 changeset = schema.changeset(target_version)
-                for version, change in changeset:
-                    log.msg('upgrading to schema version %d' % (version + 1))
-                    schema.runchange(version, change, 1)
+                with sautils.withoutSqliteForeignKeys(engine):
```

Comment:
No :-(, I've documented why in database.rst
I tried hard to make it work, but eventually sqlalchemy-migrate does not help